patches/tModLoader/Terraria/ModLoader/ModWaterStyle.cs

```diff
@@ -60,6 +61,23 @@ public abstract class ModWaterStyle : ModTexturedType
 		public virtual Color BiomeHairColor() {
 			return new Color(28, 216, 94);
 		}
+
+		/// <summary>
+		/// Returns the texture to be used when drawing rain of this water type.
+		/// <br/>Default uses the vanilla rain texture.
+		/// </summary>
+		public virtual Asset<Texture2D> GetRainTexture() {
+			return TextureAssets.Rain;
+		}
+
+		/// <summary>
+		/// Return the type of rain used. Equal to the offset in the rain texture divided by four.
```

Comment:
The use of Type here is potentially confusing. 

I would prefer we use Variant or Variation as its a varying the drawn portion of an singular texture of an existing type, not a full blown Type (ie ModType for how we use the term) in itself.

Likewise for every other usage of Type in this PR,  use Variant instead

Reply:
It's `type` in terraria code, hence the wordchoice, but I agree that `variant` is better.